pkg/artifactory/resource_artifactory_local_alpine_repository.go

```diff
@@ -4,37 +4,39 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-var alpineLocalSchema = mergeSchema(baseLocalRepoSchema, map[string]*schema.Schema{
-	"primary_keypair_ref": {
-		Type:     schema.TypeString,
-		Optional: true,
-		Description: "Used to sign index files in Alpine Linux repositories. " +
-			"See: https://www.jfrog.com/confluence/display/JFROG/Alpine+Linux+Repositories#AlpineLinuxRepositories-SigningAlpineLinuxIndex",
-	},
-}, compressionFormats)
-
 func resourceArtifactoryLocalAlpineRepository() *schema.Resource {
+	const pkt = "alpine"
+
+	var alpineLocalSchema = mergeSchema(getBaseLocalRepoSchema(pkt), map[string]*schema.Schema{
+		"primary_keypair_ref": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Description: "Used to sign index files in Alpine Linux repositories. " +
+				"See: https://www.jfrog.com/confluence/display/JFROG/Alpine+Linux+Repositories#AlpineLinuxRepositories-SigningAlpineLinuxIndex",
+		},
+	}, compressionFormats)
+
+	type AlpineLocalRepo struct {
+		LocalRepositoryBaseParams
+		PrimaryKeyPairRef string `hcl:"primary_keypair_ref" json:"primaryKeyPairRef"`
+	}
+
+	var unPackLocalAlpineRepository = func(data *schema.ResourceData) (interface{}, string, error) {
+		d := &ResourceData{ResourceData: data}
+		repo := AlpineLocalRepo{
+			LocalRepositoryBaseParams: unpackBaseRepo("local", data, "alpine"),
```

Comment:
Notice you have `alpine` here too?

Reply:
Improvement done. variable packageType used to replace literal `alpine` from all occurrences